src/PHPFUI/MySQLSlowQuery/Parser.php

```diff
@@ -138,41 +138,81 @@ private function parse() : void
 			throw new Exception\EmptyLog(self::class . ': ' . $this->fileName . ' appears to not exist or is empty');
 			}
 
+		$parseMode = '';
 		$currentSession = [];
 
+		// Get line from stack (if pushed in below loop) or file. Any comment line
+		// not starting with "# Time: " gets discarded.
 		while (\strlen($line = $this->getNextLine()))
 			{
 			if (0 === \stripos($line, self::PORT))	// in middle of session, end it
 				{
 				$currentSession[] = $line;
+				$parseMode = $this->getParseMode($currentSession[0]);
 				// eat the next line
 				$this->getNextLine();
 				// create a new session
 				$this->sessions[] = new \PHPFUI\MySQLSlowQuery\Session($currentSession);
 				$currentSession = [];
 				$this->inSession = false;
+
+				// The next line is expected to be a comment connected to the first
+				// query in this session. In non backward compatible mode, check this:
+				// if it's not a comment line, this is assumed to be the start of the
+				// next session header (i.e. there are zero queries in this session).
+				if ($parseMode === 'mariadb' && \strlen($line = $this->getNextLine()) > 0)
+					{
+					if ('#' !== $line[0])
+						{
+						$this->inSession = true;
+						}
+					$this->pushLine($line);
+					}
```

Comment:
The location for this fix is slightly awkward... which can be fixed... by reordering much of the logic in this loop.